Build failure: canaille

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. nix-build -A canaille

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/292437039

### Relevant log output

```console
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
unpacking source archive /nix/store/13zin86plpasn05gzykznawn8p5m77ni-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/uv.lock"
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built canaille-0.0.57-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for canaille-0.0.57-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
Executing pypaInstallPhase
Successfully installed canaille-0.0.57-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
shrinking RPATHs of ELF executables and libraries in /nix/store/75r0ldkad7iv7fxjgr0m1bmxz5difzmn-canaille-0.0.57
checking for references to /build/ in /nix/store/75r0ldkad7iv7fxjgr0m1bmxz5difzmn-canaille-0.0.57...
patching script interpreter paths in /nix/store/75r0ldkad7iv7fxjgr0m1bmxz5difzmn-canaille-0.0.57
stripping (with command strip and flags -S -p) in  /nix/store/75r0ldkad7iv7fxjgr0m1bmxz5difzmn-canaille-0.0.57/lib /nix/store/75r0ldkad7iv7fxjgr0m1bmxz5difzmn-canaille-0.0.57/bin
shrinking RPATHs of ELF executables and libraries in /nix/store/wy8963ix9d2snbchfyb8k260ps0gj00w-canaille-0.0.57-dist
checking for references to /build/ in /nix/store/wy8963ix9d2snbchfyb8k260ps0gj00w-canaille-0.0.57-dist...
patching script interpreter paths in /nix/store/wy8963ix9d2snbchfyb8k260ps0gj00w-canaille-0.0.57-dist
Rewriting #!/nix/store/wz0j2zi02rvnjiz37nn28h3gfdq61svz-python3-3.12.9/bin/python3.12 to #!/nix/store/wz0j2zi02rvnjiz37nn28h3gfdq61svz-python3-3.12.9
wrapping `/nix/store/75r0ldkad7iv7fxjgr0m1bmxz5difzmn-canaille-0.0.57/bin/canaille'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
Running phase: pythonRemoveBinBytecodePhase
Running phase: pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
Executing pytestCheckPhase
pytest flags: -m pytest --numprocesses=2
============================= test session starts ==============================
platform linux -- Python 3.12.9, pytest-8.3.4, pluggy-1.5.0
rootdir: /build/source
configfile: pyproject.toml
plugins: cov-6.0.0, pytest_httpserver-1.1.0, lazy-fixtures-1.1.1, smtpdfix-0.5.2, xdist-3.6.1, Faker-33.3.1, time-machine-2.15.0
2 workers [1467 items] m
........................................................................ [  4%]
............................sss......................................... [  9%]
........................................................................ [ 14%]
.......................................s................................ [ 19%]
...............s........................................................ [ 24%]
........................................................................ [ 29%]
........................................................................ [ 34%]
........................................................................ [ 39%]
........................................................................ [ 44%]
........................................................................ [ 49%]
........................................................................ [ 53%]
........................................................................ [ 58%]
........................................................................ [ 63%]
........................................................................ [ 68%]
........................................................................ [ 73%]
........................................................................ [ 78%]
..............................................FF.F...................... [ 83%]
........................................................................ [ 88%]
........................................................................ [ 93%]
........................................................................ [ 98%]
...........................                                              [100%]
=================================== FAILURES ===================================
______________________ test_invalid_client[ldap_backend] _______________________
[gw0] linux -- Python 3.12.9 /nix/store/wz0j2zi02rvnjiz37nn28h3gfdq61svz-python3-3.12.9/bin/python3.12

testclient = <flask_webtest.TestApp object at 0x7fffdb8f1f10>
backend = <canaille.backends.ldap.backend.LDAPBackend object at 0x7fffdbd889e0>
user = <User user_name=user>

    def test_invalid_client(testclient, backend, user):
        assert not testclient.app.config["CANAILLE_OIDC"].get(
            "DYNAMIC_CLIENT_REGISTRATION_OPEN"
        )
        testclient.app.config["CANAILLE_OIDC"]["DYNAMIC_CLIENT_REGISTRATION_TOKENS"] = [
            "static-token"
        ]
    
        payload = {
            "client_id": "invalid-client-id",
            "redirect_uris": ["https://newname.example.test/callback"],
        }
    
        headers = {"Authorization": "Bearer static-token"}
        res = testclient.put_json(
            "/oauth/register/invalid-client-id", payload, headers=headers, status=401
        )
>       assert res.json == {"error": "invalid_client"}
E       AssertionError: assert {'error': 'in...this server.'} == {'error': 'invalid_client'}
E         
E         Omitting 1 identical items, use -vv to show
E         Left contains 1 more item:
E         {'error_description': 'The client does not exist on this server.'}
E         Use -v to get more diff

tests/oidc/test_dynamic_client_registration_management.py:177: AssertionError
---------------------------- Captured stdout setup -----------------------------
[2025-03-13 12:18:39,140] DEBUG in selector_events: Using selector: EpollSelector
[2025-03-13 12:18:39,140] INFO in controller: SMTPDFix running on localhost:42191
[2025-03-13 12:18:39,143] DEBUG in selector_events: Using selector: EpollSelector
[2025-03-13 12:18:39,144] INFO in controller: SMTPDFix running on localhost:42191
------------------------------ Captured log setup ------------------------------
DEBUG    asyncio:selector_events.py:64 Using selector: EpollSelector
INFO     smtpdfix.controller:controller.py:71 SMTPDFix running on localhost:42191
DEBUG    asyncio:selector_events.py:64 Using selector: EpollSelector
INFO     smtpdfix.controller:controller.py:71 SMTPDFix running on localhost:42191
----------------------------- Captured stdout call -----------------------------
[2025-03-13 12:18:39,471] DEBUG in oauth: client registration management endpoint request: POST: {}
[2025-03-13 12:18:39,472] DEBUG in oauth: client registration management endpoint response: {'error': 'invalid_client', 'error_description': 'The client does not exist on this server.'}
------------------------------ Captured log call -------------------------------
DEBUG    canaille:oauth.py:268 client registration management endpoint request: POST: {}
DEBUG    canaille:oauth.py:275 client registration management endpoint response: {'error': 'invalid_client', 'error_description': 'The client does not exist on this server.'}
_____________________ test_invalid_client[memory_backend] ______________________
[gw0] linux -- Python 3.12.9 /nix/store/wz0j2zi02rvnjiz37nn28h3gfdq61svz-python3-3.12.9/bin/python3.12

testclient = <flask_webtest.TestApp object at 0x7fffe9afd490>
backend = <canaille.backends.memory.backend.MemoryBackend object at 0x7fffe9affc50>
user = <User id=844b0769-58e4-496b-8608-820fd1edd8dd>

    def test_invalid_client(testclient, backend, user):
        assert not testclient.app.config["CANAILLE_OIDC"].get(
            "DYNAMIC_CLIENT_REGISTRATION_OPEN"
        )
        testclient.app.config["CANAILLE_OIDC"]["DYNAMIC_CLIENT_REGISTRATION_TOKENS"] = [
            "static-token"
        ]
    
        payload = {
            "client_id": "invalid-client-id",
            "redirect_uris": ["https://newname.example.test/callback"],
        }
    
        headers = {"Authorization": "Bearer static-token"}
        res = testclient.put_json(
            "/oauth/register/invalid-client-id", payload, headers=headers, status=401
        )
>       assert res.json == {"error": "invalid_client"}
E       AssertionError: assert {'error': 'in...this server.'} == {'error': 'invalid_client'}
E         
E         Omitting 1 identical items, use -vv to show
E         Left contains 1 more item:
E         {'error_description': 'The client does not exist on this server.'}
E         Use -v to get more diff

tests/oidc/test_dynamic_client_registration_management.py:177: AssertionError
---------------------------- Captured stdout setup -----------------------------
[2025-03-13 12:18:39,478] DEBUG in selector_events: Using selector: EpollSelector
[2025-03-13 12:18:39,478] INFO in controller: SMTPDFix running on localhost:40493
[2025-03-13 12:18:39,481] DEBUG in selector_events: Using selector: EpollSelector
[2025-03-13 12:18:39,481] INFO in controller: SMTPDFix running on localhost:40493
------------------------------ Captured log setup ------------------------------
DEBUG    asyncio:selector_events.py:64 Using selector: EpollSelector
INFO     smtpdfix.controller:controller.py:71 SMTPDFix running on localhost:40493
DEBUG    asyncio:selector_events.py:64 Using selector: EpollSelector
INFO     smtpdfix.controller:controller.py:71 SMTPDFix running on localhost:40493
----------------------------- Captured stdout call -----------------------------
[2025-03-13 12:18:39,509] DEBUG in oauth: client registration management endpoint request: POST: {}
[2025-03-13 12:18:39,509] DEBUG in oauth: client registration management endpoint response: {'error': 'invalid_client', 'error_description': 'The client does not exist on this server.'}
------------------------------ Captured log call -------------------------------
DEBUG    canaille:oauth.py:268 client registration management endpoint request: POST: {}
DEBUG    canaille:oauth.py:275 client registration management endpoint response: {'error': 'invalid_client', 'error_description': 'The client does not exist on this server.'}
_______________________ test_invalid_client[sql_backend] _______________________
[gw0] linux -- Python 3.12.9 /nix/store/wz0j2zi02rvnjiz37nn28h3gfdq61svz-python3-3.12.9/bin/python3.12

testclient = <flask_webtest.TestApp object at 0x7fffe9afc980>
backend = <canaille.backends.sql.backend.SQLBackend object at 0x7fffdbf72b40>
user = <User user_name=user>

    def test_invalid_client(testclient, backend, user):
        assert not testclient.app.config["CANAILLE_OIDC"].get(
            "DYNAMIC_CLIENT_REGISTRATION_OPEN"
        )
        testclient.app.config["CANAILLE_OIDC"]["DYNAMIC_CLIENT_REGISTRATION_TOKENS"] = [
            "static-token"
        ]
    
        payload = {
            "client_id": "invalid-client-id",
            "redirect_uris": ["https://newname.example.test/callback"],
        }
    
        headers = {"Authorization": "Bearer static-token"}
        res = testclient.put_json(
            "/oauth/register/invalid-client-id", payload, headers=headers, status=401
        )
>       assert res.json == {"error": "invalid_client"}
E       AssertionError: assert {'error': 'in...this server.'} == {'error': 'invalid_client'}
E         
E         Omitting 1 identical items, use -vv to show
E         Left contains 1 more item:
E         {'error_description': 'The client does not exist on this server.'}
E         Use -v to get more diff

tests/oidc/test_dynamic_client_registration_management.py:177: AssertionError
---------------------------- Captured stdout setup -----------------------------
[2025-03-13 12:18:39,514] DEBUG in selector_events: Using selector: EpollSelector
[2025-03-13 12:18:39,515] INFO in controller: SMTPDFix running on localhost:43723
[2025-03-13 12:18:39,517] DEBUG in selector_events: Using selector: EpollSelector
[2025-03-13 12:18:39,517] INFO in controller: SMTPDFix running on localhost:43723
------------------------------ Captured log setup ------------------------------
DEBUG    asyncio:selector_events.py:64 Using selector: EpollSelector
INFO     smtpdfix.controller:controller.py:71 SMTPDFix running on localhost:43723
DEBUG    asyncio:selector_events.py:64 Using selector: EpollSelector
INFO     smtpdfix.controller:controller.py:71 SMTPDFix running on localhost:43723
----------------------------- Captured stdout call -----------------------------
[2025-03-13 12:18:39,568] DEBUG in oauth: client registration management endpoint request: POST: {}
[2025-03-13 12:18:39,570] DEBUG in oauth: client registration management endpoint response: {'error': 'invalid_client', 'error_description': 'The client does not exist on this server.'}
------------------------------ Captured log call -------------------------------
DEBUG    canaille:oauth.py:268 client registration management endpoint request: POST: {}
DEBUG    canaille:oauth.py:275 client registration management endpoint response: {'error': 'invalid_client', 'error_description': 'The client does not exist on this server.'}
=============================== warnings summary ===============================
../../nix/store/x4vpx9l9af9x2gd16gi042g85acspnmm-python3.12-webob-1.8.8/lib/python3.12/site-packages/webob/compat.py:5
../../nix/store/x4vpx9l9af9x2gd16gi042g85acspnmm-python3.12-webob-1.8.8/lib/python3.12/site-packages/webob/compat.py:5
../../nix/store/x4vpx9l9af9x2gd16gi042g85acspnmm-python3.12-webob-1.8.8/lib/python3.12/site-packages/webob/compat.py:5
  /nix/store/x4vpx9l9af9x2gd16gi042g85acspnmm-python3.12-webob-1.8.8/lib/python3.12/site-packages/webob/compat.py:5: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    from cgi import parse_header

tests/app/test_mails.py::test_send_test_email_without_credentials[sql_backend]
tests/app/commands/test_check.py::test_check_command[sql_backend]
  /nix/store/l3aa3djpbdd3j2dspf7df5rh9ybccnw9-python3.12-passlib-1.7.4/lib/python3.12/site-packages/passlib/utils/__init__.py:854: DeprecationWarning: 'crypt' is deprecated and slated for removal in Python 3.13
    from crypt import crypt as _crypt

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/oidc/test_dynamic_client_registration_management.py::test_invalid_client[ldap_backend] - AssertionError: assert {'error': 'in...this server.'} == {'error': 'invalid...
FAILED tests/oidc/test_dynamic_client_registration_management.py::test_invalid_client[memory_backend] - AssertionError: assert {'error': 'in...this server.'} == {'error': 'invalid...
FAILED tests/oidc/test_dynamic_client_registration_management.py::test_invalid_client[sql_backend] - AssertionError: assert {'error': 'in...this server.'} == {'error': 'invalid...
======= 3 failed, 1459 passed, 5 skipped, 5 warnings in 95.92s (0:01:35) =======
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.17-xanmod1, NixOS, 24.11 (Vicuna), 24.11.20250307.20755fa`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/kuroko/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/kuroko/.nix-profile/etc/xdg/nix/nix.conf:/home/kuroko/.local/state/nix/profile/etc/xdg/nix/nix.conf:/home/kuroko/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/kuroko/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/ys1zqxjyiczkny9bfi1f7i6clm1z39cd-gnome-settings-daemon-47.2/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/rbgz8k2289gz06xfxy7k7bc58i4dxxvc-lix-2.91.1/share`
 - nixpkgs: `/nix/store/l75sdw6vydkwaa6cfrf33h2pnzikpxm8-source`

### Notify maintainers

@erictapen 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ca/canaille/package.nix
+++ b/pkgs/by-name/ca/canaille/package.nix
@@ -120,6 +120,16 @@ python.pkgs.buildPythonApplication rec {
     };
   };
 
+  disabledTests = [
+    # cause by authlib being too up-to-date for this version of canaille
+    # see: https://github.com/NixOS/nixpkgs/issues/389861#issuecomment-2726361949
+    # FIX: update and see if this is fixed
+    "test_invalid_client[ldap_backend]"
+    "test_invalid_client[memory_backend]"
+    "test_invalid_client[sql_backend]"
+    "test_password_reset[sql_backend]"
+  ];
+
   meta = with lib; {
     description = "Lightweight Identity and Authorization Management";
     homepage = "https://canaille.readthedocs.io/en/latest/index.html";

